### PR TITLE
fix: cyclonedx check is unnecessary (only called in CI)

### PR DESCRIPTION
**Pull request by mistydemeo**
I've also adjusted what counts as a "global" build - the "all" and "host" modes build global artifacts too, so we should check those.

We discussed this and realized there was a mismatch between how we check for cyclonedx and how we use it. It's currently executed in CI, rather than from Rust as one of the builds, so we don't actually run it during `cargo build -aglobal`. As a result there's no real reason to check for it right now - it's always installed in the only place we use it. We can add back the check if we move the execution to Rust. #1652

### CODE REVIEW

**Comment by mistydemeo** [comment]
I was trying to figure out why the tests are failing with "no cyclonedx" even though the tests run fine without them, but then I realized what the issue is. We do the cyclonedx stuff in CI, not in Rust. Our test builds invoke `dist build -aglobal`, which means they don't actually call cyclonedx at all - but this check is still there.

We should either remove the cyclonedx check altogether, or we should move the command to run it into Rust so that the two sides of this agree with each other.

File: cargo-dist/src/lib.rs [hunk 1]
```diff
@@ -61,12 +61,16 @@ mod tests;
 pub fn do_env_test(cfg: &Config) -> DistResult<()> {
     let (dist, _manifest) = tasks::gather_work(cfg)?;
 
-    let global_builds = cfg.artifact_mode == ArtifactMode::Global;
+    let global_builds = matches!(
+        cfg.artifact_mode,
+        ArtifactMode::Global | ArtifactMode::All | ArtifactMode::Host
+    );
```
[the remark this answers is not in the document]
Comment (mistydemeo) [reply cargo-dist/src/lib.rs:67]: Hmmm, right. I think `!global_builds` doesn't really make sense at this point - I'll reword this to `let local_builds =` and have it match on local/host/all.

**Review by duckinator — APPROVED** [review]
